src/nn/ops/clamp.ts

```diff
@@ -88,9 +73,9 @@ export class Clamp extends SingleOutputOperation {
 
   runOp(x: tf.Tensor): tf.Tensor {
     utils.assert(
-        this.minScalarValue_ !== undefined &&
-            this.maxScalarValue_ !== undefined,
+        this.minValue_ !== undefined &&
+            this.maxValue_ !== undefined,
         'tf.js only supports clipByValue.');
-    return tf.clipByValue(x, this.minScalarValue_, this.maxScalarValue_);
+    return tf.clipByValue(x, this.minValue_, this.maxValue_);
```

Comment:
Implement the fast path by `tf.clipByValue` when `minValue` and `maxValue` are not undefined. Otherwise, use `tf.maximum` and `tf.minimum` instead.

Reply:
Thanks @huningxin, I got it.